FV3/coarse_graining/coarse_graining.F90

```diff
@@ -582,21 +585,45 @@ subroutine vertical_remapping_requirements(delp, area, ptop, phalf, upsampled_co
     deallocate(coarse_phalf)
    end subroutine vertical_remapping_requirements
 
-   subroutine mask_area_weights(area, phalf, upsampled_coarse_phalf, masked_area_weights)
+   subroutine mask_area_weights(area, phalf, upsampled_coarse_phalf, extrapolate, masked_area_weights)
     real, intent(in) :: area(is:ie,js:je)
     real, intent(in) :: phalf(is:ie,js:je,1:npz+1)
     real, intent(in) :: upsampled_coarse_phalf(is:ie,js:je,1:npz+1)
+    logical, intent(in) :: extrapolate
     real, intent(out) :: masked_area_weights(is:ie,js:je,1:npz)
 
+    real, allocatable :: upsampled_coarse_pfull(:,:,:)
```

Comment:
Is this variable used outside of extrapolate case? Can we initialize as `allocate(upsampled_coarse_pfull(is:ie,js:je,1:npz))` already here? Or is this intentional to not allocate extra memory?  Alternatively, is it okay to initialize this inside `compute_pfull_from_phalf`?

Reply:
It is only used in the extrapolation case, so I defer allocating it to the `extrapolate` branch of the code (otherwise it is not allocated).  We might be able to get away with allocating it in `compute_pfull_from_phalf`, but my instinct is to stick with the way I've done it here since this is the outermost scope in which it is used, and it allows us to preserve the explicit array sizes in the subroutine signature (consistent with the other code I wrote in this module).